client/state/posts/selectors.js

```diff
@@ -158,3 +158,20 @@ export function getSitePostsForQueryIgnoringPage( state, siteId, query ) {
 		return memo.concat( getSitePostsForQuery( state, siteId, pageQuery ) || [] );
 	}, [] );
 }
+
+/**
+ * Returns true if a request is in progress for the specified site post, or
+ * false otherwise.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Number}  postId Post ID
+ * @return {Boolean}        Whether request is in progress
+ */
+export function isRequestingSitePost( state, siteId, postId ) {
+	if ( ! state.posts.siteRequests[ siteId ] ) {
```

Comment:
>are you guaranteed that `state.posts.siteRequests` exists?

Yep, because the Redux store will be initialized with the [default return value of the reducer](https://github.com/Automattic/wp-calypso/blob/615773ac9f6c4c9d596ae3a0e990d80278d3572e/client/state/posts/reducer.js#L54). Under the hood, the initial state is constructed during the Redux internal `@@INIT` action type.